internal/app/singularity/actions.go

```diff
@@ -553,6 +554,12 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 			sylog.Verbosef("starter-suid not found, using user namespace")
 			UserNamespace = true
 		}
+
+		// check for ownership of singularity.conf file when !UserNamespace
+		if !fs.IsOwner(configurationFile, 0) {
```

Comment:
Make this an 'else if' to work correctly when starter-suid is missing.  That's the case in an unprivileged installation that I maintain.

Reply:
@ArangoGutierrez Those checks should definitively go in `PrepareConfig` function (`internal/pkg/runtime/engines/singularity/prepare.go`) to have something like this : 

```
	if starterConfig.GetIsSUID() {
		// check for ownership of singularity.conf
		if !fs.IsOwner(configurationFile, 0) {
			sylog.Fatalf("%s must be owned by root", configurationFile)
		}
	}
```

Same for `capability.json` but in `prepareUserCaps` function.

CLI won't prevent `starter-suid` from reading a configuration file not owned by root, that's why it must be done in `internal/pkg/runtime/engines/singularity/prepare.go`